script/fetch_electron.py

```diff
@@ -0,0 +1,84 @@
+import sys
+import os
+import subprocess
+
+def is_depot_tools_installed():
+  try:
+    with open(os.devnull, "w") as f:
+      subprocess.check_call(["gclient", "help"], stdout=f)
+    return True
+  except (OSError, subprocess.CalledProcessError):
+    return False
+
+def find_gclient_root(search_dir='.'):
```

Comment:
honestly i didn't know that `gclient root` was a thing. it might actually make sense to just import the gclient python library and call the `GClient.LoadCurrentConfig` function directly :)

Reply:
I looked into doing this, and I think the best option is to keep this `find_gclient_root` function, for a few reasons:

1. getting the path to `depot_tools` is nontrivial, because python 2.7 doesn't have `shutil.which`.
2. `gclient root` doesn't distinguish between 'found a root' and 'returned the current directory', and this script needs to be able to tell the difference in order to be able to tell whether it needs to run `gclient config`
3. the functions I'd call in `gclient_utils.py` aren't any kind of exposed API, so it's entirely possible that those functions will stop working or be renamed in a refactor